javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -81,6 +121,272 @@ class Properties(protected val defaults: Properties)
       defaults.foreachAncestor(f)
   }
 
-  // def list(out: PrintStream): Unit
-  // def list(out: PrintWriter): Unit
+  private final val listStr = "-- listing properties --"
+
+  private def format(entry: ju.Map.Entry[AnyRef, AnyRef]): String = {
+    val key: String   = entry.getKey().asInstanceOf[String]
+    val value: String = entry.getValue().asInstanceOf[String]
+    if (key.length > 40)
+      s"${key.substring(0, 37)}...=$value"
+    else
+      s"$key=$value"
+  }
+
+  def list(out: PrintStream): Unit = {
+    out.println(listStr)
+    entrySet().scalaOps.foreach { entry => out.println(format(entry)) }
+  }
+
+  def list(out: PrintWriter): Unit = {
+    out.println(listStr)
+    entrySet().scalaOps.foreach { entry => out.println(format(entry)) }
+  }
+
+    private def loadImpl(reader: Reader): Unit = {
+    import java.util.regex._
+    val trailingBackspace = Pattern.compile("""(\\)+$""")
+    val br                = new BufferedReader(reader)
+    val valBuf            = new jl.StringBuilder()
+    var prevValueContinue = false
+    var isKeyParsed       = false
+    var key: String       = null
+    var line: String      = null
+
+    while ({ line = br.readLine(); line != null }) {
+      var i: Int   = -1
+      var ch: Char = Char.MinValue
+
+      def getNextChar(): Char = {
+        i += 1
+        // avoid out of bounds if value is empty
+        if (i < line.length())
```

Comment:
this format is dangerous in refactorings - i'd put {} around the branches or put them in the same line as the if and else (the latter might frowned upon here)

Reply:
I am not sure about this one as it passed the Scala Native review and there may be a couple of more places too. I am also not sure about the Dotty "significant whitespace" feature either. I'll gladly change as needed either way.

Reply:
We typically do not put braces around an if/else when all of its branches are one liners. The current format is good here.